lib/puppet/defaults.rb

```diff
@@ -727,13 +727,36 @@ def self.initialize_default_settings!(settings)
         unless [0, 'unlimited', Float::INFINITY].include?(val)
           Puppet.deprecation_warning(<<-WARNING)
 Fine grained control of environment timeouts is deprecated,
-please use `0` or `unlimited` to control default caching behavior
-and the environment-cache endpoint in Puppet Server's administrative
-API to expire the cache as needed
+please use `environment_ttl` instead.
           WARNING
         end
       end
     },
+    :environment_ttl => {
+      :default    => "0",
+      :type       => :ttl,
+      :desc       => "How long the Puppet server should cache data it loads from an
+      environment based on when it was last accessed.
+
+      A value of `0` will disable caching. This setting can also be set to
+      `unlimited`, which will cache environments until the server is restarted
+      or told to refresh the cache. All other values will result in Puppet server
+      refresh environments that have not been accessed within that ttl.
```

Comment:
All other values will result in Puppet Server evicting cached environments that have not been access within the TTL?

Reply:
Yeah I was trying to keep the same "refresh" terminology belong, but reading it again "evicted" makes sense here. I'll update that.